prov/efa/src/rdm/efa_rdm_pke_rtm.c

```diff
@@ -649,36 +662,11 @@ ssize_t efa_rdm_pke_proc_matched_eager_rtm(struct efa_rdm_pke *pkt_entry)
 		err = efa_rdm_pke_copy_payload_to_ope(pkt_entry, rxe);
 		if (err)
 			efa_rdm_pke_release_rx(pkt_entry);
-
-		return err;
-	}
-
-	/* In this case, data is already in user provided buffer, so no need
-	 * to copy. However, we do need to make sure the packet header length
-	 * is correct. Otherwise, user will get wrong data.
-	 *
-	 * The expected header size is
-	 * 	ep->msg_prefix_size - sizeof(struct efa_rdm_pke)
-	 * because we used the first sizeof(struct efa_rdm_pke) to construct
-	 * a pkt_entry.
-	 */
-	hdr_size = pkt_entry->payload - pkt_entry->wiredata;
-	if (hdr_size != pkt_entry->ep->msg_prefix_size - sizeof(struct efa_rdm_pke)) {
-		/* if header size is wrong, the data in user buffer is not useful.
-		 * setting rxe->cq_entry.len here will cause an error cq entry
-		 * to be written to application.
-		 */
-		rxe->cq_entry.len = 0;
 	} else {
-		rxe->cq_entry.len = pkt_entry->pkt_size + sizeof(struct efa_rdm_pke);
+		efa_rdm_pke_rtm_handle_zcpy_recv_completion(pkt_entry, rxe);
```

Comment:
why do you still need to handle zcpy recv in this function?

Reply:
I figure the function is just poorly named. Is it safe to remove the logic completely from `efa_rdm_pke_proc_matched_eager_rtm()`?

Reply:
I mean you already make a branch in `efa_rdm_pke_handle_recv_completion` that the zcpy received pkts should not enter `efa_rdm_proc_received` , who calls `efa_rdm_pke_proc_matched_eager_rtm`. Then we can remove this logic safely here, right?

But Brian is doubting if this shortcut is the best approach, we can discuss that.